eng/tools/mcp/azure-sdk-java-mcp/src/update-java-sdk.ts

```diff
@@ -0,0 +1,102 @@
+import { CallToolResult } from "@modelcontextprotocol/sdk/types";
+import { spawnAsync } from "./utils/index.js";
+import * as fs from "fs";
+import * as path from "path";
+
+export async function updateJavaSdk(
+  packagePath: string,
+  commitHash?: string,
+  repo?: string,
+  tspConfig?: string,
+  localSpec?: string,
+): Promise<CallToolResult> {
+  try {
+    // Validate package path exists and contains tsp-location.yaml
+    if (!fs.existsSync(packagePath)) {
+      return {
+        content: [
+          {
+            type: "text",
+            text: `Error: Package path does not exist: ${packagePath}`,
+          },
+        ],
+      };
+    }
+
+    const tspLocationPath = path.join(packagePath, "tsp-location.yaml");
+    if (!fs.existsSync(tspLocationPath)) {
+      return {
+        content: [
+          {
+            type: "text",
+            text: `Error: tsp-location.yaml not found in ${packagePath}. This appears to be a new project - use sync_java_sdk instead.`,
+          },
+        ],
+      };
+    }
+
+    // Build TypeSpec arguments for update command
+    const args = ["update", "--debug"];
```

Comment:
tsp-client update does not have `--local-spec-repo` parameter, so I think it is not supported. I think we still need to call init command with `--local-spec-repo` if we want to differtiate existing and new sdk. But I did not see benifit of it though.

https://github.com/Azure/azure-sdk-tools/tree/main/tools/tsp-client#update
https://github.com/Azure/azure-sdk-tools/blob/main/tools/tsp-client/src/commands.ts#L344-L347